diff --git a/toolchain/llvm/llvm.bzl b/toolchain/llvm/llvm.bzl
index 2010c623..8879de3a 100644
--- a/toolchain/llvm/llvm.bzl
+++ b/toolchain/llvm/llvm.bzl
@@ -15,7 +15,10 @@ def declare_llvm_targets(*, suffix = ""):
     )
-
+
     # Convenient exports
-    native.exports_files(native.glob(["bin/*"]))
+    native.exports_files(
+        native.glob(["bin/*"]),
+        visibility = ["//visibility:public"],
+    )
-
+
     native.filegroup(
         name = "clangxx_file",
