diff --git a/rust/private/repository_utils.bzl b/rust/private/repository_utils.bzl
index 7c0f5f1..c4f1bf0 100644
--- a/rust/private/repository_utils.bzl
+++ b/rust/private/repository_utils.bzl
@@ -380,6 +380,7 @@ rust_toolchain(
     process_wrapper = "@rules_rust//util/process_wrapper",
     linker = {linker_label},
     linker_type = {linker_type},
+    rust_objcopy = {rust_objcopy_label},
     rustfmt = {rustfmt_label},
     cargo = "//:cargo",
     clippy_driver = "//:clippy_driver_bin",
